py/torch_tensorrt/dynamo/conversion/impl/interpolate.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional, Sequence
+
+import tensorrt as trt
+from torch.fx.node import Target
+from torch_tensorrt.dynamo._SourceIR import SourceIR
+from torch_tensorrt.dynamo.conversion._ConversionContext import ConversionContext
+from torch_tensorrt.fx.converters.converter_utils import set_layer_name
+from torch_tensorrt.fx.types import TRTTensor
+
+
+def interpolate(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    out_shape: Optional[Sequence[int]],
+    scale_factor: Sequence[float],
+    interpolate_mode: Optional[str],
+    align_corners: Optional[bool],
+) -> TRTTensor:
+    resize_layer = ctx.net.add_resize(input)
+    # output size calculation
+    # Pytorch assures that only one of out_shape/scale_factor is not None
+    # Pytorch assures that dimensions match for out_shape/scale factor
+    if out_shape is not None:
+        resize_layer.shape = list(input.shape)[:2] + list(out_shape)
+    else:
+        resize_layer.scales = [1, 1] + list(scale_factor)
```

Comment:
Should both branches of the conditional ensure that both `shape` and `scales` are set, or can TRT handle cases where only one is set?

Reply:
@gs-olive 
Looks like there are some internal assertions in Pytorch that checks the arguments for `shape` and `scales`. 
So pytorch will throws out an error if none of them is set or both of them are set. 
Since this is a converter so we assumes that the arguments are valid or pytorch will throw out error right? 
Please correct me if I'm wrong, thanks!